llvm/lib/CodeGen/MachineFunctionPass.cpp

```diff
@@ -91,7 +96,15 @@ bool MachineFunctionPass::runOnFunction(Function &F) {
 
   MFProps.reset(ClearedProperties);
 
-  bool RV = runOnMachineFunction(MF);
+  bool RV;
+  if (DroppedVarStatsMIR) {
+    auto PassName = getPassName();
+    DroppedVarStatsMF.runBeforePass(PassName, &MF);
+    RV = runOnMachineFunction(MF);
+    DroppedVarStatsMF.runAfterPass(PassName, &MF);
```

Comment:
Looking at this usage, why is it necessary to store DroppedVarStatsMF on the MachineFunction? Isn't it possible to instantiate it here?

Reply:
Ah yes, you are correct, that will work, thanks for the suggestion!

Reply:
https://llvm-compile-time-tracker.com/compare.php?from=5e59fa57d29e159f41957958ba541049f85652d6&to=94f4cf922deb8997ff10525b58432cf2eab8bee2&stat=instructions:u 

It is down to 0.04%